File: android/app/src/main/java/de/repeatuntil/yata/infrastructure/todolist/TodoListMemoryRepository.kt
package de.repeatuntil.yata.infrastructure.todolist

import de.repeatuntil.yata.domain.common.Id
import de.repeatuntil.yata.domain.todolist.TodoList
import de.repeatuntil.yata.domain.todolist.TodoListRepository

class TodoListMemoryRepository : TodoListRepository {

    private val todoLists = mutableListOf<TodoList>()

    override suspend fun getTodoListForUser(userId: Id): Result<TodoList?> {
        return Result.success(todoLists.firstOrNull { it.userId == userId })
    }

    override suspend fun addTodoList(todoList: TodoList): Result<Unit> {
        todoLists.add(todoList)
        return Result.success(Unit)
    }

    override suspend fun updateTodoList(todoList: TodoList): Result<Unit> {
        val index = todoLists.indexOfFirst { it.id == todoList.id }
        if (index >= 0) {
            todoLists[index] = todoList
        }
        return Result.success(Unit)
    }
}

File: ios/YATA/Infrastructure/TodoList/TodoListMemoryRepository.swift
//
//  TodoListMemoryRepository.swift
//  YATA
//
//  Created by Alpár Szotyori on 26.05.24.
//

import Foundation

class TodoListMemoryRepository: TodoListRepository {

    private var todoLists: [TodoList] = []

    func getTodoListForUser(_ userId: YATA.Id) async -> Result<YATA.TodoList?, any Error> {
        return Result.success(todoLists.first { $0.userId == userId })
    }

    func addTodoList(_ todoList: YATA.TodoList) async -> Result<Void, any Error> {
        todoLists.append(todoList)
        return Result.success(())
    }

    func updateTodoList(_ todoList: YATA.TodoList) async -> Result<Void, any Error> {
        if let index = todoLists.firstIndex(where: { $0.id == todoList.id }) {
            todoLists[index] = todoList
        }
        return Result.success(())
    }

}
